volatility3/framework/plugins/linux/bash_env.py

```diff
@@ -0,0 +1,227 @@
+# This file is Copyright 2022 Volatility Foundation and licensed under the Volatility Software License 1.0
+# which is available at https://www.volatilityfoundation.org/license/vsl-v1.0
+#
+
+import struct
+from typing import List
+
+from volatility3.framework import constants, renderers, symbols, interfaces
+from volatility3.framework.configuration import requirements
+from volatility3.framework.interfaces import plugins
+from volatility3.framework.objects import utility
+from volatility3.framework.symbols.linux.hash import HashIntermedSymbols
+from volatility3.plugins.linux import pslist
+
+
+class Var(plugins.PluginInterface):
+    """Recovers a process' dynamic environment variables."""
+
+    _required_framework_version = (2, 0, 0)
+
+    @classmethod
+    def get_requirements(cls) -> List[interfaces.configuration.RequirementInterface]:
+        return [
+            requirements.ModuleRequirement(
+                name="kernel",
+                description="Linux kernel",
+                architectures=["Intel32", "Intel64"],
+            ),
+            requirements.PluginRequirement(
+                name="pslist", plugin=pslist.PsList, version=(2, 0, 0)
+            ),
+            requirements.ListRequirement(
+                name="pid",
+                element_type=int,
+                description="Process IDs to include (all other processes are excluded)",
+                optional=True,
+            ),
+        ]
+
+    def _generator(self, tasks):
+        vmlinux = self.context.modules[self.config["kernel"]]
+        is_32bit = not symbols.symbol_table_is_64bit(
+            self.context, vmlinux.symbol_table_name
+        )
+        if is_32bit:
+            pack_format = "I"
+            hash_json_file = "hash32"
+        else:
+            pack_format = "Q"
+            hash_json_file = "hash64"
+        for task in tasks:
+            proc_layer_name = task.add_process_layer()
+            if not proc_layer_name:
+                yield (0, (task.pid, utility.array_to_string(task.comm), ""))
```

Comment:
@ikelos should this empty string be NotApplicableValue instead? If the process doesn't have a layer then its likely a kernel thread, which never has environment variables.

Reply:
Yep, good spot thanks!  Anywhere that a value isn't applicable, it's best to indicate that with a specific type, so that other interfaces (or programs) can understand and handle it better.